classes/WP_CLI/Action/Create.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+use function \WP_CLI\Utils\get_flag_value;
+
+/**
+ * WP-CLI command: action-scheduler action create
+ */
+class ActionScheduler_WPCLI_Action_Create_Command extends ActionScheduler_WPCLI_Command {
+
+	const ASYNC_OPTS = array( 'async', 'now', 0 );
+
+	/**
+	 * Execute command.
+	 *
+	 * @uses as_schedule_single_action()
+	 * @uses as_enqueue_async_action()
+	 * @uses as_schedule_recurring_action()
+	 * @uses as_schedule_cron_action()
+	 * @uses $this->print_error()
+	 * @uses $this->print_success()
+	 * @return void
+	 */
+	public function execute() {
+		$hook           = $this->args[0];
+		$schedule_start = $this->args[1];
+		$callback_args  = get_flag_value( $this->assoc_args, 'args', array() );
+		$group          = get_flag_value( $this->assoc_args, 'group', '' );
+		$interval       = absint( get_flag_value( $this->assoc_args, 'interval', 0 ) );
+		$cron           = get_flag_value( $this->assoc_args, 'cron', '' );
+
+		if ( ! empty( $callback_args ) ) {
+			$callback_args = json_decode( $callback_args, true );
+		}
+
+		$function_args = array(
+			'start'         => 'async',
+			'cron'          => $cron,
+			'interval'      => $interval,
+			'hook'          => $hook,
+			'callback_args' => $callback_args,
+			'group'         => $group,
+		);
+
+		// Generate schedule start if appropriate.
+		if ( ! in_array( $schedule_start, static::ASYNC_OPTS, true ) ) {
+			$schedule_start         = as_get_datetime_object( $schedule_start );
+			$function_args['start'] = $schedule_start->format( 'U' );
+		}
+
+		// Default to creating single action.
+		$action_type = 'single';
+		$function    = 'as_schedule_single_action';
+
+		if ( 'async' === $function_args['start'] ) { // Enqueue async action.
+			$action_type = 'async';
+			$function    = 'as_enqueue_async_action';
+
+			$function_args = array_filter(
+				$function_args,
+				static function( $key ) {
+					return in_array( $key, array( 'hook', 'callback_args', 'group' ), true );
+				},
+				ARRAY_FILTER_USE_KEY
+			);
+
+		} else if ( ! empty( $interval ) ) { // Creating recurring action.
+			$action_type = 'recurring';
+			$function    = 'as_schedule_recurring_action';
+
+		} else if ( ! empty( $cron ) ) { // Creating cron action.
+			$action_type = 'cron';
+			$function    = 'as_schedule_cron_action';
+		}
+
+		$function_args = array_values( array_filter( $function_args ) );
+
+		try {
+			$actions_added = call_user_func_array( $function, $function_args );
```

Comment:
this is not the number of actions added, but instead the action ID. I suppose there will always be only one action created in this command anyway. And would be `0` if there was no action created, we should also print_error when the returning ID is 0.